components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -361,183 +322,194 @@ void Nudging::run_impl (const double dt)
-    auto tmp_state_field = get_helper_field(name+"_tmp"); // ext horiz, int vert
-    auto ext_state_view  = ext_state_field.get_view<mPack**>();
-    auto tmp_state_view  = tmp_state_field.get_view<mPack**>();
-    auto atm_state_view  = atm_state_field.get_view<mPack**>();  // TODO: Right now assume whatever field is defined on COLxLEV
-    auto int_state_view  = int_state_field.get_view<mPack**>();
-    auto int_mask_view = m_buffer.int_mask_view;
-    // Masked values in the source data can lead to strange behavior in the vertical interpolation.
-    // We pre-process the data and map any masked values (sometimes called "filled" values) to the
-    // nearest un-masked value.
-    // Here we are updating the ext_state_view, which is the time interpolated values taken from the nudging
-    // data.
     Real var_fill_value = constants::DefaultFillValue<Real>().value;
     // Query the helper field for the fill value, if not present use default
-    if (ext_state_field.get_header().has_extra_data("mask_value")) {
-      var_fill_value = ext_state_field.get_header().get_extra_data<float>("mask_value");
+    if (f.get_header().has_extra_data("mask_value")) {
+      var_fill_value = f.get_header().get_extra_data<float>("mask_value");
     }
-    const int num_cols           = ext_state_view.extent(0);
-    const int num_vert_packs     = ext_state_view.extent(1);
-    const int num_src_levs       = m_num_src_levs;
-    const auto policy = ESU::get_default_team_policy(num_cols, num_vert_packs);
-    Kokkos::parallel_for("correct_for_masked_values", policy,
-       	       KOKKOS_LAMBDA(MemberType const& team) {
-      const int icol = team.league_rank();
-      auto ext_state_view_1d = ekat::subview(ext_state_view,icol);
-      Real fill_value;
-      int  fill_idx = -1;
-      // Scan top to surf and backfill all values near TOM that are masked.
-      for (int kk=0; kk<num_src_levs; ++kk) {
-        const auto ipack = kk / mPack::n;
-	const auto iidx  = kk % mPack::n;
-        // Check if this index is masked
-	if (ext_state_view_1d(ipack)[iidx]!=var_fill_value) {
-	  fill_value = ext_state_view_1d(ipack)[iidx];
-	  fill_idx = kk;
-	  for (int jj=0; jj<kk; ++jj) {
-            const auto jpack = jj / mPack::n;
-	    const auto jidx  = jj % mPack::n;
-	    ext_state_view_1d(jpack)[jidx] = fill_value;
-	  }
-	  break;
-	}
-      }
-      // Now fill the rest, the fill_idx should be non-negative.  If it isn't that means
-      // we have a column that is fully masked 
-      for (int kk=fill_idx+1; kk<num_src_levs; ++kk) {
-        const auto ipack = kk / mPack::n;
-	const auto iidx  = kk % mPack::n;
-        // Check if this index is masked
-	if (!(ext_state_view_1d(ipack)[iidx]==var_fill_value)) {
-	  fill_value = ext_state_view_1d(ipack)[iidx];
-	} else {
-	  ext_state_view_1d(ipack)[iidx] = fill_value;
-	}
+
+    const int ncols = fl.dim(0);
+    const int nlevs = fl.dim(1);
+    const auto thresh = std::abs(var_fill_value)*0.0001;
```

Comment:
Just to understand better, is this step because now that we are not vertically interpolating first we may find ourselves time interpolating between a Filled and UnFilled value?

Reply:
This is done _before_ horiz, so we're in the same situation as before. The pb is that time interp _may_ generate something not equal to FillValue, even if both endpoints are FillValue (b/c of rounding).

Reply:
The bigger question, which I didn't consider here, is what to do if one value is FillValue and the other is not FillValue. The time interp will return something that is quite far from FillValue. We should probably move the handling of FillValue _inside_ the time interp routine.

Reply:
yeah, that was my thought too.

Reply:
Should we push changing time interpolation to a separate PR so that we are not holding up this one?